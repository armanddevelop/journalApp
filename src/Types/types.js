export const types = {
  login: "[Auth] Login",
  logout: "[Auth] LogOut",
  uiSetError: "[UI] setError",
  uiRemoveError: "[UI] removeError",
  uiStartLoading: "[UI] Start loading",
  uiFinishLoading: "[UI] Finish loading",
  uiErrorInFirebase: "[UI] Error logIn",
  uiRemoveErrorInFireBase: "[UI] Remove Error logIn",
};
