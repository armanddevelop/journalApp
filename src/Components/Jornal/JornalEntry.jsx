import moment from "moment";
import PropTypes from "prop-types";
import { useDispatch } from "react-redux";
import { notesActiveAction } from "../../Actions/notes";

export const JornalEntry = ({ title, body, date, url, id }) => {
  const noteDate = moment(date);
  const dispatch = useDispatch();
  const handleEntryClick = () => {
    const noteObj = { title, body, date, url };
    dispatch(notesActiveAction(id, noteObj));
  };
  return (
    <div className="jornal__entry pointer" onClick={handleEntryClick}>
      {url && (
        <div
          className="jornal__entry-picture"
          style={{
            backgroundSize: "cover",
            backgroundImage: `url(${url})`,
          }}
        ></div>
      )}
      <div className="jornal__entry-body">
        <p className="jornal__entry-title">{title}</p>
        <p className="jornal__entry-content">{body}</p>
      </div>
      <div className="jornal__entry-date-box">
        <span>{noteDate.format("dddd")}</span>
        <h4>{noteDate.format("Do")}</h4>
      </div>
    </div>
  );
};

JornalEntry.propTypes = {
  title: PropTypes.string.isRequired,
  body: PropTypes.string.isRequired,
  date: PropTypes.number.isRequired,
  id: PropTypes.string.isRequired,
  url: PropTypes.string,
};
