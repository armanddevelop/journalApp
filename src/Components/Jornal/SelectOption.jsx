export const SelectOption = () => {
  return (
    <div className="nothing__main-content">
      <p>
        Select an Item
        <br />
        or Create a new Entry
      </p>
      <i className="far fa-star fa-4x mt-5"></i>
    </div>
  );
};
