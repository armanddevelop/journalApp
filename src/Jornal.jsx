import { AppRouter } from "../src/Routers/AppRouter";

export const Jornal = () => {
  return (
    <div>
      <AppRouter />
    </div>
  );
};
