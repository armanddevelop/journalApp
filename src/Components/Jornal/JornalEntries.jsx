import { JornalEntry } from "./JornalEntry";

export const JornalEntries = () => {
  const entries = [1, 2, 3, 4, 5];
  return (
    <div className="jornal__entries">
      {entries.map((entry) => (
        <JornalEntry key={entry} />
      ))}
    </div>
  );
};
