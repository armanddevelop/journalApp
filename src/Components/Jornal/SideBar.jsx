import { useDispatch, useSelector } from "react-redux";
import { startLogOut } from "../../Actions/auth";
import { startNewNoteAction } from "../../Actions/notes";
import { JornalEntries } from "./JornalEntries";

export const SideBar = () => {
  const dispatch = useDispatch();
  const { name } = useSelector((state) => state.auth);
  const handleLogOut = () => {
    dispatch(startLogOut());
  };
  const handleNewEntry = () => {
    console.log("licha");
    dispatch(startNewNoteAction());
  };
  return (
    <aside className="jornal__sidebar">
      <div className="jornal__sidebar-navbar">
        <h3 className="mt-5">
          <i className="far fa-moon"></i>
          <span> {name}</span>
        </h3>
        <button className="btn" onClick={handleLogOut}>
          LogOut
        </button>
      </div>
      <div className="jornal__new-entry" onClick={handleNewEntry}>
        <i className="far fa-calendar-plus fa-5x jornal__new-entry-icon"></i>
        <p className="mt-5">New Entry</p>
      </div>
      <JornalEntries />
    </aside>
  );
};
