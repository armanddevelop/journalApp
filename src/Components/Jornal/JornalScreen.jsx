import { SelectOption } from "./SelectOption";
import { useSelector } from "react-redux";
import { NoteScreen } from "../Notes/NoteScreen";
import { SideBar } from "./SideBar";
export const JornalScreen = () => {
  const { active } = useSelector((state) => state.notes);
  return (
    <div className="jornal__main-content">
      <SideBar />
      <main>{!active ? <SelectOption /> : <NoteScreen />}</main>
    </div>
  );
};
