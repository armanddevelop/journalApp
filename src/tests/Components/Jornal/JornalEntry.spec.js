import { mount } from "enzyme";
import { Provider } from "react-redux";
import configureStore from "redux-mock-store";
import thunk from "redux-thunk";
import { JornalEntry } from "../../../Components/Jornal/JornalEntry";
import { types } from "../../../Types/types";

const title = "Component";
const body = "This is the body";
const date = 1649197509169;
const id = "123456";
const initState = {
  auth: {},
  ui: {
    errorMsgFb: {
      error: false,
      msgError: "",
    },
  },
  notes: {
    active: {
      id: "safdbdg32435",
    },
    notes: [],
    saveNote: {
      isSave: true,
      message: "All rigth",
      title: "note save",
    },
  },
};
const middlewares = [thunk];
const mockStore = configureStore(middlewares);
let store = mockStore(initState);
describe("Test in JornalEntry compoent", () => {
  const wrapper = mount(
    <Provider store={store}>
      <JornalEntry title={title} body={body} date={date} id={id} />
    </Provider>
  );
  it("SnapShot to JornalEntry component without url prop", () => {
    expect(wrapper).toMatchSnapshot();
    expect(wrapper.find(".jornal__entry-picture").exists()).toEqual(false);
  });

  it("SnapShot to JornalEntry component with url prop", () => {
    const url = "http://www.thisisUrl.com";
    const wrapper = mount(
      <Provider store={store}>
        <JornalEntry title={title} body={body} date={date} id={id} url={url} />
      </Provider>
    );

    expect(wrapper).toMatchSnapshot();
    expect(wrapper.find(".jornal__entry-picture").exists()).toEqual(true);
  });

  it("Should execute handleEntryClick", () => {
    wrapper.find(".jornal__entry").first().prop("onClick")();
    const actions = store.getActions();
    const objectNote = {
      id: expect.any(String),
      title: expect.any(String),
      body: expect.any(String),
      date: expect.any(Number),
      url: undefined,
    };

    expect(actions[0].type).toEqual(types.notesActive);
    expect(actions[0].payload).toEqual(objectNote);
  });
});
