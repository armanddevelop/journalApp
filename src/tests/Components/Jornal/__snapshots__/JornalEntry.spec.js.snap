// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Test in JornalEntry compoent SnapShot to JornalEntry component with url prop 1`] = `
<div
  className="jornal__entry pointer animate__animated animate__fadeIn animate__faster"
  onClick={[Function]}
>
  <div
    className="jornal__entry-picture"
    style={
      Object {
        "backgroundImage": "url(http://www.thisisUrl.com)",
        "backgroundSize": "cover",
      }
    }
  />
  <div
    className="jornal__entry-body"
  >
    <p
      className="jornal__entry-title"
    >
      Component
    </p>
    <p
      className="jornal__entry-content"
    >
      This is the body
    </p>
  </div>
  <div
    className="jornal__entry-date-box"
  >
    <span>
      Tuesday
    </span>
    <h4>
      5th
    </h4>
  </div>
</div>
`;

exports[`Test in JornalEntry compoent SnapShot to JornalEntry component without url prop 1`] = `
<div
  className="jornal__entry pointer animate__animated animate__fadeIn animate__faster"
  onClick={[Function]}
>
  <div
    className="jornal__entry-body"
  >
    <p
      className="jornal__entry-title"
    >
      Component
    </p>
    <p
      className="jornal__entry-content"
    >
      This is the body
    </p>
  </div>
  <div
    className="jornal__entry-date-box"
  >
    <span>
      Tuesday
    </span>
    <h4>
      5th
    </h4>
  </div>
</div>
`;
